File: Introduction_to_numerical_programming_using_Python_and_CPP_Beu/Ch06/Python/P06-CompareZeros.py
# Compare solvers for algebraic and transcendental equations
from math import *
from roots import *

def func(x): return (x - sin(x) - 0.25e0)

# main

a = -30e0; b = 30e0                                         # search interval
(x,ierr,it) = Bisect(func,a,b)              # add it to return list in Bisect
if (ierr == 0):
   print('Bisect: x = {0:8.5f}   f(x) = {1:7.0e}   it = {2:d}'.
         format(x,func(x),it))

x = -30e0                                             # initial approximation
(x,ierr,it) = Secant(func,a,b,x)            # add it to return list in Secant
if (ierr == 0):
   print('Secant: x = {0:8.5f}   f(x) = {1:7.0e}   it = {2:d}'.
         format(x,func(x),it))

File: Introduction_to_numerical_programming_using_Python_and_CPP_Beu/Ch06/C/P06-CompareZeros.cpp
// Compare solvers for algebraic and transcendental equations
#include <stdio.h>
#include <math.h>
int it;       // comment out "int it;" in Bisect and Secant to make it global
#include "roots.h"

double func(double x) { return x - sin(x) - 0.25e0; }

int main()
{
   double a, b, x;

   a = -30e0; b = 30e0;                                    // search interval
   if (Bisect(func,a,b,x) == 0)
      printf("Bisect: x = %11.9f   f(x) = %7.0e   it = %i\n",x,func(x),it);

   x = -30e0;                                        // initial approximation
   if (Secant(func,a,b,x) == 0)
      printf("Secant: x = %11.9f   f(x) = %7.0e   it = %i\n",x,func(x),it);
}
